docs/coordinates.rst

```diff
@@ -0,0 +1,444 @@
+
+.. _global_coords:
+
+GlobalCoords
+============
+Sometimes coordinates are not associated with any axis.
+Take the case of a 2-D `~ndcube.NDCube` representing a single image.
+The time at which that image was taken is important piece of coordinate information.
+But because the data does not have a 3rd dimension, it cannot be stored in the WCS or `~ndcube.ExtraCoords` objects.
+Storing such coordinates is the role of the `ndcube.GlobalCoords` class.
+`~ndcube.NDCube` is instatiated with an empty `~ndcube.GlobalCoords` object already attached at `ndcube.NDCube.global_coords`.
+Coordinates can be added to this object if and when the user sees fit.
+Let's attach a scalar global coordinate to ``my_cube`` representing some kind of distance.
+We do this by supplying the coordinate's name, physical type and value via the `~ndcube.GlobalCoords.add` method.
+
+.. code-block:: python
+
+  >>> import astropy.units as u
+  >>> my_cube.global_coords.add('distance', 'pos.distance', 1 * u.m)
+
+Because `~ndcube.GlobalCoords` allows multiple coordinates of the same physical type, a unique coordinate name must be provided.
+Furthermore the physical type must be a valid `IVOA UCD1+ controlled words <http://www.ivoa.net/documents/REC/UCD/UCDlist-20070402.html>`_ word.
+If one does not exist for your coordinate, prepend the type with ``custom:``.
+
+The value of the coordinate can be accessed by indexing the `~ndcube.GlobalCoords` instance with the coordinate name.
+
+.. code-block:: python
+
+  >>> my_cube.global_coords['distance']
+  <Quantity 1. m>
+
+The coordinate's physical type can be accessed via the `~ndcube.GlobalCoords.physical_types` `dict` property.
+
+.. code-block:: python
+
+  >>> my_cube.global_coords.physical_types['distance']
+  'pos.distance'
+
+Because `~ndcube.GlobalCoords` inherits from `~collections.abc.Mapping`, it contains a number of mixin methods similar to those of `dict`.
+
+.. code-block:: python
+
+  >>> list(my_cube.global_coords.keys())  # Returns a list of global coordinate names
+  ['distance']
+  >>> list(my_cube.global_coords.values())  # Returns a list of coordinate values
+  [<Quantity 1. m>]
+  >>> list(my_cube.global_coords.items())  # Returns a list of (name, value) pairs
+  [('distance', <Quantity 1. m>)]
+
+A common use case for `~ndcube.GlobalCoords` is associated with slicing (:ref:`cube_slicing`).
+In addition to tracking and updating the `~ndcube.NDCube.wcs` and `~ndcube.NDCube.extra_coords` objects, `~ndcube.NDCube`'s slicing infrastucture also identifies when the array axes to which a coordinate corresponds are dropped.
+The values of dropped coordinates at the position where the `~ndcube.NDCube` was sliced are stored in the `astropy.wcs.WCS` instance from where `~ndcube.GlobalCoords` can access and return them.
+
+.. code-block:: python
```

Comment:
if only this example actually worked :stuck_out_tongue:

Reply:
Indeed!